File: 2206-Divide-Array-Into-Equal-Pairs/2206.cpp
class Solution {
public:
    bool divideArray(vector<int>& nums) {

        size_t len = nums.size();
        if(len % 2 != 0) return false;

        for(int i=0; i<len; i++) {
            if(nums[i]==-1) continue;
            for(int j=i+1; j<len; j++) {
                if(nums[i]==nums[j]){
                    nums[i]=-1;
                    nums[j]=-1;
                    break;
                }
            }
            if(nums[i]!=-1) return false;
        }   

        for(int i=0; i<len; i++) {
            if(nums[i]!=-1) return false;
        }

        return true;
    }
};

File: 2206-Divide-Array-Into-Equal-Pairs/2206.rs
impl Solution {
    pub fn divide_array(nums: Vec<i32>) -> bool {
        let mut count = vec![0u16; 500];

        for num in nums {
            count[num as usize - 1] += 1;
        } 

        for i in count {
            if i%2 != 0 {
                return false;
            }
        }

        true
    }
}
